add third party signin button to hamburger menu

JaydenTeoh opened the pull request:
<!-- What issue does this PR close? -->
Closes #6937 

<!-- What does this PR achieve? [feature|hotfix|fix|refactor] -->
Added third party login button under hamburger menu

### Screenshot
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->
<img width="297" alt="My Open Library" src="https://user-images.githubusercontent.com/90945854/221422121-182e1987-7606-4ffa-9ac4-af6b97a567c5.png">

### Stakeholders
<!-- @ tag stakeholders of this bug -->
@cdrini created issue

mekarpeles commented:
@cdrini my concern is around any sort of iframe tracking across the website. Do you mind investigating?

cdrini commented:
We'll need someway to show/hide the iframe depending on if the hamburger menu has been opened. We want to keep these pieces relatively decoupled, so don't want the iframe eg listening to an event. Instead would prefer the parent component, the header dropdown, to call the iframe when it sees fit.

We can do this by having some global function inside the ia_third_party_logins iframe to act as an API, and then from the header_dropdown, we can listen to the [toggle event](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#events) on the details element, and call the methods as needed

JaydenTeoh commented:
@cdrini @mekarpeles I have implemented lazy loading of the iframe using window events but I am unable to unload/change the `src` attribute of the iframe to an empty string upon closing the hamburger menu. I believe this is due to the violation of the frame-ancestors directive of the implemented Content Security Policy (CSP). Is there any way around this?

cdrini commented on openlibrary/templates/account/ia_thirdparty_logins.html:
```diff
@@ -1,12 +1,24 @@
 ></iframe>
+
+<script>
+    window.addEventListener('LOAD_THIRD_PARTY_LOGINS', function() {
+        var iframe = document.getElementById('ia-third-party-logins');
+        iframe.src = iframe.getAttribute('data-src')
+    });
+
+    window.addEventListener('UNLOAD_THIRD_PARTY_LOGINS', function() {
+        var iframe = document.getElementById('ia-third-party-logins');
+        iframe.src = '';
+    });
```
Let's define these as globals:
```suggestion
    window.LOAD_THIRD_PARTY_LOGINS = function() {
        var iframe = document.getElementById('ia-third-party-logins');
        iframe.src = iframe.getAttribute('data-src');
    };

    window.UNLOAD_THIRD_PARTY_LOGINS = function() {
        var iframe = document.getElementById('ia-third-party-logins');
        iframe.src = '';
    };

```

Then call these functions directly with eg `window.LOAD_THIRD_PARTY_LOGINS()`. The event handling creates an extra level of misdireciton, and uses some things which would require polyfilling (eg `Event`)

JaydenTeoh commented:
@cdrini I have made the changes requested and now the `UNLOAD_THIRD_PARTY_LOGINS` function is also successful in changing the HTML src attribute of the iframe to an empty string! Not sure why it works now after changing to global functions even though I still get the CSP warning in the console.

Anyways, here are the changes I made according to your comments:

1. Change loading and unloading into global window functions
2. Added a new class to the details element in `header_dropdown.html` (`<details class="$(props['name'])-details">`)
3. Added the following watch function in `openlibrary/js/index.js`:
```
// Load and unload third party login iframes when toggling hamburger menu
    const hamburgerMenu = document.querySelector('.hamburger-details')
    hamburgerMenu.addEventListener('toggle', () => {
        if (hamburgerMenu.open) {
            window.LOAD_THIRD_PARTY_LOGINS()
        } else {
            window.UNLOAD_THIRD_PARTY_LOGINS()
        }
    })
```

Do let me know if I missed out on anything!

cdrini reviewed (approved):
Niiice! Working smoothly :) A few small changes. Also pushing some change to archive.org side to make the iframe a little nicer:

![image](https://user-images.githubusercontent.com/6251786/225772646-afd5fee1-8c73-4b5b-9974-fbe7c1e23c7a.png)

cdrini commented:
This looks good! Will merge once the archive.org changes are merged through :+1:

cdrini commented:
Deployed on IA! :+1:
